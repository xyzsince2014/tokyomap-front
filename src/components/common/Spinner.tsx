import * as React from 'react';
import Loader from 'react-loader-spinner';

const Spinner: React.FC = () => (
  <div className="l-spinner">
    <Loader type="ThreeDots" color="#333" />
  </div>
);

export default Spinner;
