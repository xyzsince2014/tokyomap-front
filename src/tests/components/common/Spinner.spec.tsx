import {create} from 'react-test-renderer';

// in case of using enzyme
// import {configure} from 'enzyme';
// import * as EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';

import Spinner from '../../../components/common/Spinner';

describe('Spinner component', () => {
  // in case of using enzyme
  // configure({
  //   adapter: new EnzymeAdapter(),
  //   disableLifecycleMethods: true,
  // });

  it('snapshot', () => {
    const tree = create(<Spinner />);
    expect(tree).toMatchSnapshot();
  });
});
