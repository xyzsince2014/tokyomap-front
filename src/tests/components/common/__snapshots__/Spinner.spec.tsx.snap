// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component snapshot 1`] = `
<div
  className="l-spinner"
>
  <div
    aria-busy="true"
    className=""
    style={Object {}}
  >
    <svg
      aria-label="audio-loading"
      fill="#333"
      height={80}
      viewBox="0 0 120 30"
      width={80}
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="15"
        cy="15"
        r={15}
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
      <circle
        attributeName="fillOpacity"
        cx="60"
        cy="15"
        from="1"
        r={9}
        to="0.3"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="9"
          repeatCount="indefinite"
          to="9"
          values="9;15;9"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="0.5"
          repeatCount="indefinite"
          to="0.5"
          values=".5;1;.5"
        />
      </circle>
      <circle
        cx="105"
        cy="15"
        r={15}
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
